### ISSUE TITLE
Order in #set

### ISSUE THREAD

**Issue by askurashev** [SemanticMediaWiki/SemanticMediaWiki#4293]
MediaWiki 1.31.3
PHP 7.0.33-0+deb9u3 (apache2handler)
MariaDB 10.1.41-MariaDB-0+deb9u1
Semantic MediaWiki 3.0.2

Hello.
After update SMW from 2.5* to 3.0.2 I noticed an unpleasant feature for myself.
1. In in an earlier version:
```
{{Publication
|Authors    = Pélabon C., Thöne P., Hansen T.F., Armbruster W.S.
}}

{{#set:|PAuthors={{{Authors|}}}|+sep}}

{{#ask: [[{{PAGENAME}}]] | ?PAuthors|mainlabel=-|headers=hide|link=all}}
```

And I get: Pélabon C., Thöne P., Hansen T.F., Armbruster W.S. 

Pay attention to the order of the authors!

2. In 3.0.2 SMW version:

Code is similar.

And I get: Armbruster W.S., Pélabon C., Thöne P., Hansen T.F

Pay attention to the order of the authors!

It is important for me that the order in which the authors were listed in the template is preserved even when they are displayed.

Why did the output change?

**Comment by jaideraf** [comment]
@askurashev before SMW 3.1.0 (not released yet), SMW could not guarantee the sequence of the values assignment to be the same as the values displayed. More context is presented here: https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/4226

This is fixed in 3.1.0, please see this [help page about sequence map](https://www.semantic-mediawiki.org/wiki/Help:Sequence_map).

SMW 3.1.0 is about to be released in a few weeks (I suppose). For now, you can test this feature in the [release candidate 3.1.0-rc.1](https://sourceforge.net/p/semediawiki/mailman/message/36755208/).

**Comment by askurashev** [comment]
Thanks for the information.
